include/channel_table.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * @file channel_table.h
+ * @brief Pr
+ */
+
+#ifndef INCLUDE_CHANNEL_TABLE_H_
+#define INCLUDE_CHANNEL_TABLE_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "charbuf.h"
+
+typedef struct ChannalKeyTable
+{
+  charbuf *chan_key;
+  size_t num_entries;
+} ChanTable;
+
+extern ChanTable channal_table;
```

Comment:
This doesn't match `channel_table.c`.
It's also unused, but maybe you plan to use it later on.

```suggestion
extern ChanTable chan_table;
```